src/libecalc/common/time_utils.py

```diff
@@ -122,6 +137,11 @@ def get_period(self, time: datetime) -> Period:
 
         raise ProgrammingError(f"Period for date '{time}' not found in periods")
 
+    @property
+    def period(self) -> Period:
+        """Get the period which is the union of all the sub periods"""
+        return Period(start=self.periods[0].start, end=self.periods[-1].end)
```

Comment:
do we know that periods is at least of length 1? I can see that previous methods make same assumption...